htdocs/lib2/logic/attribute.class.php

```diff
@@ -143,4 +143,24 @@ public static function getAttributesListArrayInternal($cacheId, $bOnlySelectable
 
         return $attributes;
     }
+
+    /**
+     * @param $attribId
+     * @return array
+     */
+    public static function getConflictingAttribIds($attribId)
+    {
+        static $conflicts = [];
+
+        if (!$conflicts) {
+            $rs = sql('SELECT `attrib1_id`, `attrib2_id` FROM `cache_attrib_conflicts`');
+            while ($r = sql_fetch_array($rs)) {
+                $conflicts[$r[0]][] = $r[1];
+                $conflicts[$r[1]][] = $r[0];
+            }
+            sql_free_result($rs);
+        }
```

Comment:
it is not easier to store the conflicted attributes directly as an array in this method?

Reply:
Yes. I considered that and even implemented it, and then went for the database table, becaus it is more flexible: It allows to use this information directly in SQL commands.

But if it's important to you, I can change it back to the internal array. The code then may need another rewrite if we eventually need the information in SQL.

Reply:
Btw, comments should be added to the table to make the ID numbers easier to understand. So I would either add them as code comments, or as an additional DB column, before this PR is merged.

Reply:
I prefer an internal array structure.

Comments sounds good :+1:

Reply:
Have moved the table to PHP code. I suggest to keep the old commit, so that the code can be picked from Git history if needed.